sig-and-wg/sig-core/proposals/generating-changelog-with-pr-labels.md

```diff
@@ -0,0 +1,105 @@
+
+**Example:**
+
+> ### Installation
+> - Sample change 1
+> - Sample change 2
+>
+> ### Security
+> - Sample change 1
+> - Sample change 2
+
+In future we can introduce nested grouping for changelog (first level: change type, second one: area of the change). Change type will marked with additional prefixed labels. Choosing right prefix will be discussed later.
+
+**Example:**
+
+> ### Bug fixes
+> **Installation**
+> - Sample change 1
+> - Sample change 2
+>
+> **Security**
+> - Sample change 1
+> - Sample change 2
+>
+> ### Features
+> **Installation**
+> - Sample change 1
+> - Sample change 2
+>
+> **Security**
+> - Sample change 1
+> - Sample change 2
+
+## Pros
+
+- One place to define general changes in pull request: pull request title, which is usually prefilled with commit message, if there is just only one commit
+- Selecting type of change is a no-brainer: contributor expands the list of available PR labels and chooses right category.
+- We can filter unlabeled pull requests ([link](https://github.com/kyma-project/kyma/issues?q=is%3Aopen+is%3Apr+no%3Alabel)) and easily add labels to them to generate complete changelog. Also, it's easy to see the full list of PRs to check if they are named correctly (and adjust titles - we can do that even for closed pull requests)
+- It enforces user to make a single type of change in a pull request
+- There are many tools available for generating changelog from pull requests labels.
+
+## Cons
+
+- More detailed changes in changelog. For example, when there is a bigger feature, which needs multiple PRs, it will be visible as multiple bullet points.
+- Pull request authors and reviewers can forgot about adding a label to categorize the change. Anyway, even closed pull requests can be edited before generating the changelog. 
+
+## Tools
+
+The changelog can be generated during CI build job. There are a lot of tools for that. 
+
+#### [Lerna Changelog](https://github.com/lerna/lerna-changelog)
+
+- Very simple tool
+- Generates changelog with PR links and authors to standard output ([example](https://github.com/lerna/lerna-changelog/releases)) - we can easily pipe it to a CHANGELOG.md file and as GitHub release
+- Allows to define custom labels
+- Can be run with parameters to for example generate a full changelog
+- Uses GitHub API
+
+#### [GitHub Changelog Generator](https://github.com/github-changelog-generator/github-changelog-generator)
+
+- Generates changelog based on GitHub issues and PRs
+- Many configuration options
+- No custom labels for PRs are supported: there are only 3 sections in generated changelog: `bug`, `enhancement` and all other changes
+- Uses GitHub API
+- Written in Ruby - it can be difficult to contribute to it
```

Comment:
Could this - being based on issues and PRs - be a way to mitigate the con of multiple related PRs showing up as multiple changes?

Reply:
Hmm, it might work, but then we would need to turn the idea upside-down - generate changelog based on closed issues, not pull requests. I haven't seen this approach anywhere, but it's interesting. Anyway, let's see if the multiple PRs for changes are really an issue for us 🙂